src/components/ConfigEditor/ConfigEditor.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useEffect } from "react";
+import useSWR from "swr";
+import { getHintOptions } from "vitessce";
+
+import { StudyIdInput } from "../StudyIdInput";
+import { baseJson, exampleConfig } from "../../utils/config-examples.js";
+import { ErrorDiv } from "../ErrorDiv";
+
+import styles from "./ConfigEditor.module.css";
+
+import {
+	ERROR_MESSAGES,
+	EXAMPLE_URL,
+	LINK_ID_ENDPOINT_URL,
+	LINK_ID_LENGTH,
+	LINK_ID_KEY,
+} from "../../utils/constants.js";
+import {
+	validateConfig,
+	sanitizeURLs,
+	studyIdFetcher,
+	updateConfigWithExampleURL,
+} from "../../utils/utility-functions.js";
+
+export const ConfigEditor = ({
+	pendingJson,
+	setPendingJson,
+	error,
+	setError,
+	setLoading,
+	setUrl,
+	setLinkIdInput,
+}) => {
+	const [datasetUrls, setDatasetUrls] = useState("");
+	const [generateConfigError, setGenerateConfigError] = useState(null);
+	const [studyId, setStudyId] = useState(null);
+	const [linkId, setLinkId] = useState(null);
+	const [linkIdUrl, setLinkIdUrl] = useState(null);
+	const [showReset, setShowReset] = useState(null);
+
+	function handleGetLinkId(studyId) {
+		setStudyId(studyId);
+	}
+
+	useSWR(linkIdUrl, studyIdFetcher, {
```

Comment:
Ah, okay - this provides some context re: the reasoning for the approach used in the other component.

A pattern we've used in the portal UI in similar cases where there are multiple loading states at once is setting up a separate const which combines the loading clauses. e.g.:

```tsx

const { data: firstData, isLoading: firstDataIsLoading} = useSWR(...)
const { data: secondData, isLoading: secondDataIsLoading} = useSWR(...)

const isLoading = firstDataIsLoading || secondDataIsLoading;

```

This tends to be a bit more bulletproof than manually managing a single loading state in `onError`/`onSuccess` handlers.

Reply:
This is a good tip, thanks for sharing. In this case the two useSWR are in different files/components. For now I removed the loading from linkId as it's not needed.